src/flash.cu

```diff
@@ -1,384 +1,326 @@
+  params.k_row_stride = k_row_stride;
+  params.v_batch_stride = v_batch_stride;
+  params.v_head_stride = v_head_stride;
+  params.v_row_stride = v_row_stride;
+  params.o_batch_stride = o_batch_stride;
+  params.o_head_stride = o_head_stride;
+  params.o_row_stride = o_row_stride;
+
+  params.window_size_left = window_size_left;
+  params.window_size_right = window_size_right;
+
+  return params;
+}
+
+void flash_attention_forward(half* q_ptr, half* k_ptr, half* v_ptr, half* output_ptr,
+                             int batch_size, int seqlen_q, int seqlen_k, int num_heads,
+                             int num_heads_k, int head_dim, int q_batch_stride, int k_batch_stride,
+                             int v_batch_stride, int o_batch_stride, int q_head_stride,
+                             int k_head_stride, int v_head_stride, int o_head_stride,
+                             int q_row_stride, int k_row_stride, int v_row_stride, int o_row_stride,
+                             float softmax_scale, bool is_causal, int window_size_left,
+                             int window_size_right, cudaStream_t stream) {
+  auto params = get_fwd_params(
+      q_ptr, k_ptr, v_ptr, output_ptr, batch_size, seqlen_q, seqlen_k, num_heads, num_heads_k,
+      head_dim, q_batch_stride, k_batch_stride, v_batch_stride, o_batch_stride, q_head_stride,
+      k_head_stride, v_head_stride, o_head_stride, q_row_stride, k_row_stride, v_row_stride,
+      o_row_stride, softmax_scale, is_causal, window_size_left, window_size_right);
+  run(params, stream);
+}
+
+void flash_attention_var_len_forward(half* q_ptr, half* k_ptr, half* v_ptr, const int* cu_seqlens_q,
+                                     const int* cu_seqlens_k, half* output_ptr, int batch_size,
+                                     int max_seqlen_q, int max_seqlen_k, int num_heads,
+                                     int num_heads_k, int head_dim, int q_head_stride,
+                                     int k_head_stride, int v_head_stride, int o_head_stride,
+                                     int q_row_stride, int k_row_stride, int v_row_stride,
+                                     int o_row_stride, float softmax_scale, bool is_causal,
+                                     int total_q, int total_k, int window_size_left,
+                                     int window_size_right, cudaStream_t stream) {
+  auto params = get_fwd_params(
+      q_ptr, k_ptr, v_ptr, output_ptr, batch_size, max_seqlen_q, max_seqlen_k, num_heads,
+      num_heads_k, head_dim, 0, 0, 0, 0,  // batch strides
+      q_head_stride, k_head_stride, v_head_stride, o_head_stride, q_row_stride, k_row_stride,
+      v_row_stride, o_row_stride, softmax_scale, is_causal, window_size_left, window_size_right);
+
+  params.cu_seqlens_q = cu_seqlens_q;
+  params.cu_seqlens_k = cu_seqlens_k;
+  params.is_seqlens_k_cumulative = true;
+  params.total_q = total_q;
+  params.total_k = total_k;
+  
+  int tile_count_semaphore[1];

```

Comment:
why not just

```
int tile_count_semaphore = 0;
```
?

It's value doesn't matter for non-causal case, so it can always be set to 0.